src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/attn_quant.cpp

```diff
@@ -307,12 +485,116 @@ static void paged_attn_quant_mt(const ov::intel_cpu::PlainTensor& k_src,
     size_t B = k_src.m_dims[0], H = k_src.m_dims[1], L1 = k_src.m_dims[2], S = k_src.m_dims[3], SV = v_src.m_dims[3];
-    size_t block_size = k_dst.m_dims[2];
+    size_t block_size = quant_key_by_channel ? k_dst.m_dims[2] - 2 * sizeof(float) : k_dst.m_dims[2];
     size_t sub_byte_multiplier = 8 / v_dst.get_precision().bitwidth();
+    if (quant_key_by_channel) {
+        parallel_for2d(past_lens.size(0), H, [&](size_t sub_seq_id, size_t h) {
+            auto past_len = past_lens.ptr<int32_t>()[sub_seq_id];
+            float* buffer = temp_buffer.ptr<float>(parallel_get_thread_num());
+            auto q_len =
+                subsequence_begins.ptr<int32_t>()[sub_seq_id + 1] - subsequence_begins.ptr<int32_t>()[sub_seq_id];
+            auto block_number_start = block_indices_begins.ptr<int32_t>()[sub_seq_id];
+            auto total_blocks =
+                block_indices_begins.ptr<int32_t>()[sub_seq_id + 1] - block_indices_begins.ptr<int32_t>()[sub_seq_id];
+            size_t m = 0;
+            if (past_len == 0) {
+                parallel_for(total_blocks, [&](int32_t block_count) {
+                    auto block_id = block_number_start + block_count;
+                    auto block_number = block_indices.ptr<int32_t>()[block_id];
+                    auto token_num = (block_id == (block_indices_begins.ptr<int32_t>()[sub_seq_id + 1] - 1))
+                                         ? (q_len - (block_id - block_number_start) * block_size)
+                                         : block_size;
+                    size_t b_in_tokens = subsequence_begins.ptr<int32_t>()[sub_seq_id] + block_count * block_size;
+                    auto p_scales = reinterpret_cast<float*>(
+                        k_dst.ptr<typename ov::element_type_traits<KEY_DST_PREC>::value_type>(block_number, h, 0, 0));
+                    auto p_zps = p_scales + S;
+                    auto p_k = k_dst.ptr<typename ov::element_type_traits<KEY_DST_PREC>::value_type>(block_number,
+                                                                                                     h,
+                                                                                                     2 * sizeof(float),
+                                                                                                     0);
+                    quant_u8_by_channel_kernel(k_src.ptr<T>(b_in_tokens, h, m),
+                                               p_k,
+                                               token_num,
+                                               S,
+                                               k_src.stride(0),
+                                               k_dst.stride(2),
+                                               p_scales,
+                                               p_zps);
+                });
+            } else {
+                auto block_number = block_indices.ptr<int32_t>()[block_number_start + past_len / block_size];
```

Comment:
Does this branch miss a loop dimension about `q_len`? For example: the input length of first token is 512, the first infer will process 256 tokens with no kvcache then the second process the remaining with kvcache length of 256.